pulsar-broker/src/main/java/org/apache/pulsar/broker/loadbalance/impl/ModularLoadManagerImpl.java

```diff
@@ -1139,6 +1143,29 @@ public void writeBrokerDataOnZooKeeper(boolean force) {
         }
     }
 
+    /**
+     * sort bundles by load and select topK bundles for each broker.
+     * @return the number of bundles selected
+     */
+    private CompletableFuture<Integer> selectTopKBundle() {
```

Comment:
Do we need to apply the `topK` selection on each broker's bundle stats report stage?

Reply:
Maybe the reason is we can't get topK bundles based on topK bundles of each broker?
From this point, it looks like makes sense to introduce a new minimum throughput limitation for the bundle stats that are reported by each broker?

Reply:
The original logic is that each follower only uploads its own broker data, the leader reads and extracts BundleData, aggregates and uploads BundleData.
If the follower broker only uploads topK from the source, it can make the brokerData smaller. But I feel that such optimization may not be very significant? The pressure should mainly be caused by the reading and writing of a large number of Znodes, with a large number of Bundles corresponding to a large number of Znodes.
Moreover, I feel that the current approach also has a benefit, which is to let the leader broker select the **global** topK * brokerCount, without the need for each follower to select a **local** top K.

Reply:
I think we can scope out the brokerData optimization in this PR. Then, we can raise another PR to put only topk bundles in brokerData if further optimization is required.

Reply:
Sorry, i don't get your point. Do you agree with that leader filter topK, or follower filter topK?

Reply:
I still support that it is leader's responsibility to filter topK, bacause the main pressure is caused by large number of znodes corresponding to large bundles stats **uploaded by leader**. 
Filtering bundles in follower only minimize the single size of one znode, which do not reduce too much pressure on zk. If neccessary, we can raise another pr to implement it.